### ISSUE TITLE
Cannot convert value to floating point (bad value) when using 0 or 0.0 as the floating point value on table.insert

### ISSUE THREAD

**Issue by chriolant** [googleapis/nodejs-bigquery#45]
#### Environment details

  - OS: Running this in Cloud Functions
  - Node.js version: Running this in Cloud Functions
  - npm version: Running this in Cloud Functions
  - @google-cloud/bigquery version: 0.11.1

#### Steps to reproduce

I've been passing a field in a CSV to Natural Language API to fetch sentiments and then insert them to BigQuery, however, there are cases where users do not enter any text and I want to do an insert anyway with the sentiment scores being zero. If I try manually setting them to zero however:

```
	row["Sentimentscore"] = 0.0;
	row["Sentimentmagnitude"] = 0.0;
```

I get a bad value error. I've tried using several variations and I get the same error: 0, 0.0, ''. I know that a manual load into bigquery with a value of 0.0 works, but I can't get it to work here which perplexes me. All other values returned from NL API which aren't zero are inserted with no errors.

My insert code:

```js
bigquery
  .dataset(datasetId)
  .table(tableId)
  .insert(row,options)
  .then(() => {
    console.log('Inserted : ', JSON.stringify(row));
  })
  .catch(err => {
    if (err && err.name === 'PartialFailureError') {
      if (err.errors && err.errors.length > 0) {
          console.log('Insert errors:');
          err.errors.forEach(err => console.error(err));
    }
    } else {
    console.error('ERROR:', err);
    }
  });
```

_P.S. first time submitting an issue in a repo, apologies in advance if I made any mistakes._

**Comment by stephenplusplus** [comment]
Would you mind correcting the following snippet so that it reproduces the error?

```js
table.create({
  schema: {
    fields: [
      {
        name: 'double',
        type: 'FLOAT64',
      },
    ],
  },
}, function() {
  table.insert({
    double: 0.0
  }, function(err) {
    // Error should appear here.
  });
});```

**Comment by chriolant** [comment]
Thanks for the reply! :) The below should produce the error:

```
const schema = "field1:float";
const tableId = "float_test";

 const options = {
    schema: schema,
  };

.createTable(tableId, options); 

var row = JSON.parse('{float1:0}');

function() {
  .table(tableId)
  .insert(row)
  }, function(err) {
    // Error should appear here.
  });
});
```

**Comment by stephenplusplus** [comment]
That actually works for me, although I had to change the name of `float1` to match the field name, `field1`.

```js
table.create({ schema: 'field1:float' }, function(err) {
  var row = JSON.parse('{"field1":0}');

  table.insert(row, function(err) {
    // err === null
  });
});
```

**Comment by chriolant** [comment]
You know what... I just debugged my JSON field by field using the example you had me correct and it was actually another field that was passing a '' (empty string) to a field in BigQuery with the datatype float that was triggering the error. 

I've added some steps to detect empty strings being passed to integer, boolean and float fields to stop this from happening. Many apologies for wasting your time 😞 (although you asking me to correct the code led me to the solution at the end). 

Thank you for leading me to the right path.
